### PR TITLE
text-minimessage: Reduce the API surface of the parse context a little bit

### PR DESCRIPTION
**Pull request by zml2008**
Can more of these methods be removed? The only truly essential one is imo `parse(String)`, but I'm not sure what sort of usecases others might have

### CODE REVIEW

**Comment by kashike** [comment]
It's definitely easier to add methods as-required, versus deprecating and removing methods later on.

File: text-minimessage/src/main/java/net/kyori/adventure/text/minimessage/Context.java [hunk 1]
```diff
@@ -23,226 +23,23 @@
...
-   * Returns the root element.
-   *
-   * @return root
-   * @since 4.10.0
-   */
-  public ElementNode tokens() {
-    return this.root;
-  }
-
+public interface Context {
```
Comment (kezz) [inline text-minimessage/src/main/java/net/kyori/adventure/text/minimessage/Context.java:35]: Should we annotate this with a non-extendible notice?

**Review by kezz — CHANGES_REQUESTED** [review]

File: text-minimessage/src/main/java/net/kyori/adventure/text/minimessage/Context.java [hunk 2]
```diff
@@ -23,226 +23,25 @@
...
-  /**
-   * Returns callback ran at the end of parsing which could be used to compact the output.
-   *
-   * @return Post-processing function
-   * @since 4.10.0
-   */
-  public UnaryOperator<Component> postProcessor() {
-    return this.postProcessor;
-  }
+  String originalMessage();
```
Comment (kezz) [inline text-minimessage/src/main/java/net/kyori/adventure/text/minimessage/Context.java:44]:
```suggestion
  @NotNull String originalMessage();
```

File: text-minimessage/src/main/java/net/kyori/adventure/text/minimessage/Context.java [hunk 3]
```diff
@@ -251,7 +50,5 @@ public MiniMessage miniMessage() {
    * @return the parsed message
    * @since 4.10.0
    */
-  public Component parse(final String message) {
-    return this.miniMessage.deserialize(message, this.placeholderResolver);
-  }
+  Component parse(String message);
```
Comment (kezz) [inline text-minimessage/src/main/java/net/kyori/adventure/text/minimessage/Context.java:53]:
```suggestion
  @NotNull Component parse(final @NotNull String message);
```

**Review by kezz — CHANGES_REQUESTED** [review]

**Review by kezz — APPROVED** [review]

**Pull request merged**